src/execution/as_abi.rs

```diff
@@ -352,92 +394,108 @@ pub(crate) fn assembly_script_append_data_for(
-            sub_remaining_gas_with_mult(env, data.len(), settings::metering_get_data_value_mult())?;
-            Ok(pointer_from_bytearray(env, &data)?.offset() as i32)
+            sub_remaining_gas_with_mult(&env, &mut ctx, data.len(), settings::metering_get_data_value_mult())?;
+            Ok(pointer_from_bytearray(&env, memory, &mut ctx,&data)?.offset() as i32)
         }
         Err(err) => abi_bail!(err),
     }
 }
 
 /// Deletes a datastore entry for an address. Fails if the entry or address does not exist.
 pub(crate) fn assembly_script_delete_data_for(
-    env: &ASEnv,
+    mut ctx: FunctionEnvMut<ASEnv>,
     address: i32,
     key: i32,
 ) -> ABIResult<()> {
-    sub_remaining_gas(env, settings::metering_delete_data_const())?;
+
+    let env = ctx.data().clone();
+    sub_remaining_gas(&env, &mut ctx, settings::metering_delete_data_const())?;
     let memory = get_memory!(env);
-    let address = get_string(memory, address)?;
-    let key = read_buffer_and_sub_gas(env, memory, key, settings::metering_delete_data_key_mult())?;
+    let address = get_string(memory, &mut ctx, address)?;
+    let key = read_buffer_and_sub_gas(&mut ctx, memory, key, settings::metering_delete_data_key_mult())?;
     match env.get_interface().raw_delete_data_for(&address, &key) {
         Ok(_) => Ok(()),
         Err(err) => abi_bail!(err),
     }
 }
 
-pub(crate) fn assembly_script_has_data_for(env: &ASEnv, address: i32, key: i32) -> ABIResult<i32> {
-    sub_remaining_gas(env, settings::metering_has_data_const())?;
+pub(crate) fn assembly_script_has_data_for(mut ctx: FunctionEnvMut<ASEnv>, address: i32, key: i32) -> ABIResult<i32> {
+    let env = ctx.data().clone();
+    sub_remaining_gas(&env, &mut ctx, settings::metering_has_data_const())?;
     let memory = get_memory!(env);
-    let address = get_string(memory, address)?;
-    let key = read_buffer_and_sub_gas(env, memory, key, settings::metering_has_data_key_mult())?;
+    let address = get_string(memory, &mut ctx, address)?;
+    let key = read_buffer_and_sub_gas(&mut ctx, memory, key, settings::metering_has_data_key_mult())?;
     match env.get_interface().has_data_for(&address, &key) {
         Ok(true) => Ok(1),
         Ok(false) => Ok(0),
         Err(err) => abi_bail!(err),
     }
 }
 
-pub(crate) fn assembly_script_get_owned_addresses_raw(env: &ASEnv) -> ABIResult<i32> {
-    sub_remaining_gas(env, settings::metering_get_owned_addrs())?;
+pub(crate) fn assembly_script_get_owned_addresses_raw(mut ctx: FunctionEnvMut<ASEnv>) -> ABIResult<i32> {
+
+    let env = ctx.data().clone();
+    sub_remaining_gas(&env, &mut ctx, settings::metering_get_owned_addrs())?;
     let data = match env.get_interface().get_owned_addresses() {
         Ok(data) => data,
         Err(err) => abi_bail!(err),
     };
-    match StringPtr::alloc(&data.join(";"), env.get_wasm_env()) {
+    let memory = get_memory!(env);
+    match StringPtr::alloc(&data.join(";"), env.get_wasm_env(), memory, &mut ctx) {
```

Comment:
As said in https://github.com/massalabs/as-ffi-bindings/pull/8#discussion_r1048432865, bindings allocs should definitely only take the env, we are getting memory from the env to send it separately when we already send the env

Reply:
done